server/service/src/invoice/inbound_shipment/update/generate.rs

```diff
@@ -306,6 +335,21 @@ pub fn generate_lines_and_stock_lines(
             currency_rate,
         )?;
 
+        line.donor_id = match input.donor_update_method.clone() {
+            Some(UpdateDonorMethod::NoChanges) | None => line.donor_id,
+            Some(UpdateDonorMethod::All) => input.default_donor_id.clone(),
+            Some(UpdateDonorMethod::Existing) => {
+                if line.donor_id.is_none() {
+                    None
+                } else {
+                    input.default_donor_id.clone()
+                }
+            }
+            Some(UpdateDonorMethod::Unspecified) => {
```

Comment:
I'm thinking now of changing from 'Unspecified' to 'unassigned'. Let me know if you think naming isn't clear